src/common/Throttle.cc

```diff
@@ -368,3 +368,61 @@ void OrderedThrottle::complete_pending_ops() {
     ++m_complete_tid;
   }
 }
+
+uint32_t DynamicThrottle::calc_wait_time(uint32_t percentage_empty, 
+                  double &wait_time, bool return_start_delay_if_none) {
+
+  bool need_to_delay = false;
+  if (percentage_empty > 100) {
+    return 1;
```

Comment:
You only return 0 and 1, and the only caller ignores it...  If the user passes in a percentage about 100, should you assert?  Is the return value of this method actually void?  If so, you could just return the wait time directly and not mess around with an out param.